### PR TITLE
Fix actors with AttackFollow moving away from their targets on amove

### LINKED ISSUE #19845: Broken attack move for vehicel with turret weapon
When a vehicel with a turret weapon get another attack move order while attacking a unit in range, then it will move further across the attack move line until it lost track of the target. Then the vehicel will stop, if another target is in range.  This includes Tesla Tanks Mammoths, Rangers and so on.

https://user-images.githubusercontent.com/54892539/146076908-346defd1-d295-4fbc-8de2-fb8e8486cf75.mp4

yes .... my exchange graphic card cant handle OBS :´(

### PR DESCRIPTION
**Pull request by abcdefg30**
Fixes #19845. @UppsAlert and @CJure already tested the first version of this fix. (The second commit renaming things is new.)

### CODE REVIEW

**Review by PunkPun — COMMENTED** [review]
I tested this out, ran a big AI game and nothing seems to be broken or crashing. The bug seems to be mostly fixed, however there's a peculiar interaction with force firing you own units. 

When you force fire anything that you own, and then attack move your units will not execute that attack move, rather they will just stop killing your troops. In this scenario attack move = stop command

**Comment by abcdefg30** [comment]
Rebased. I couldn't reproduce the force firing issue with rocket soldiers firing on rifle men in TD. Can you try again and confirm that the rebase fixed the issue?

**Comment by PunkPun** [comment]
> Rebased. I couldn't reproduce the force firing issue with rocket soldiers firing on rifle men in TD. Can you try again and confirm that the rebase fixed the issue?

You need to use turreted vehicles, and yes the rebase didn't fix it

**Comment by abcdefg30** [comment]
Fixed.

**Review by PunkPun — DISMISSED** [review]
lgtm, I haven't looked at the code though

**Comment by abcdefg30** [comment]
Fixed a typo.

File: OpenRA.Mods.Common/Traits/AutoTarget.cs [hunk 1]
```diff
@@ -229,8 +229,9 @@ void INotifyDamage.Damaged(Actor self, AttackInfo e)
 			if (attacker.Disposed)
 				return;
 
-			foreach (var dat in disableAutoTarget)
-				if (dat.DisableAutoTarget(self))
+			// Don't change targets when there is a target overriding auto-targeting
+			foreach (var dat in overrideAutoTarget)
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/AutoTarget.cs]: `dat` &rarr; `oat` or `override`.

File: OpenRA.Mods.Common/Traits/AutoTarget.cs [hunk 2]
```diff
@@ -283,9 +284,9 @@ public Target ScanForTarget(Actor self, bool allowMove, bool allowTurn, bool ign
 		{
 			if ((ignoreScanInterval || nextScanTime <= 0) && ActiveAttackBases.Any())
 			{
-				foreach (var dat in disableAutoTarget)
-					if (dat.DisableAutoTarget(self))
-						return Target.Invalid;
+				foreach (var dat in overrideAutoTarget)
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/AutoTarget.cs]: Likewise here.

**Review by pchote — COMMENTED** [review]
Can we please also squash the first two commits together?

(note: initial code review only, may have more comments coming)

**Comment by abcdefg30** [comment]
Updated.

**Review by PunkPun — APPROVED** [review]

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/d47d78c9b4a7e9abe8995ab9f0460e6b84b9147e...e37940abb88eeebb62f28b3c37f7f927b77022b1)

**Pull request merged**